lib/user_data_parser/data_set/data_set_parser.rb

```diff
@@ -0,0 +1,20 @@
+require_relative "./data_set.rb"
+
+module UserDataParser
+  class DataSetParser
+    attr_reader :row_parser
+
+    def initialize(row_parser)
+      @row_parser = row_parser
+    end
+
+    def parse(data)
+      data_set = DataSet.new
+      parsed_rows = data.map do |row|
+        row_parser.parse(row)
+      end
+      data_set = data_set.add_rows(parsed_rows)
```

Comment:
Nitpick: this reads a little to me like we're altering a ```DataSet``` that already has rows. Why not just return ```DataSet.new(parsed_rows)```?